cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -128,7 +128,7 @@ func (s *server) addArchiveStorage(opts *querysvc.QueryServiceOptions, host comp
 	return nil
 }
 
-func (s *server) createMetricReader(host component.Host) (metricsstore.Reader, error) {
+func (s *server) createMetricReader(opts *querysvc.QueryServiceOptions, host component.Host) (metricsstore.Reader, error) {
```

Comment:
these `opts` are not relevant to the business of this function. You can do the assignment in the caller instead.